src/systemd/sd-event.h

```diff
@@ -69,7 +69,7 @@ typedef int (*sd_event_handler_t)(sd_event_source *s, void *userdata);
 typedef int (*sd_event_io_handler_t)(sd_event_source *s, int fd, uint32_t revents, void *userdata);
 typedef int (*sd_event_time_handler_t)(sd_event_source *s, uint64_t usec, void *userdata);
 typedef int (*sd_event_signal_handler_t)(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata);
-#if defined __USE_POSIX199309 || defined __USE_XOPEN_EXTENDED
+#if defined _GNU_SOURCE || defined _POSIX_C_SOURCE && _POSIX_C_SOURCE >= 199309L
```

Comment:
_POSIX_C_SOURCE is not defined by the libc, but it can be defined through the command line (just like _GNU_SOURCE).
siginfo_t is declared if the user requested it by defining either _POSIX_C_SOURCE or _GNU_SOURCE.